src/scripts/records/UserRecords.js

```diff
@@ -3,29 +3,106 @@
 import Immutable from 'immutable'
 import Cookies from 'js-cookie'
 
-import { REQUEST_STATUS } from 'constants/ApplicationConstants'
+import {
+  REQUEST_STATUS,
+  DEFAULT_WALLET_KEY,
+  DEFAULT_MNEMONIC_KEY
+} from 'constants/ApplicationConstants'
 
 import type { RequestStatus } from 'types/ApplicationTypes'
 
-export const _getWalletKey = (state: User): string => state.get('walletKey')
-export const _getMnemonicKey = (state: User): string => state.get('mnemonicKey')
-export const _getLoginRequestStatus = (state: User): RequestStatus =>
-  state.get('loginRequestStatus')
+class KdfParams extends Immutable.Record({
+  dklen: -1,
+  salt: '',
+  n: -1,
+  r: -1,
+  p: -1
+}) {
+  dklen: number
+  salt: string
+  n: number
+  r: number
+  p: number
+}
+
+class CipherParams extends Immutable.Record({
+  iv: ''
+}) {
+  iv: string
+}
+
+class Crypto extends Immutable.Record({
+  ciphertext: '',
+  cipherparams: new CipherParams(),
+  cipher: '',
+  kdf: '',
+  kdfparams: new KdfParams(),
+  mac: ''
+}) {
+  ciphertext: string
+  cipherparams: CipherParams
+  cipher: string
+  kdf: string
+  kdfparams: KdfParams
+  mac: string
+
+  constructor ({ cipherparams, kdfparams, ...rest }: Object = {}) {
+    super({
+      ...rest,
+      cipherparams: new CipherParams(cipherparams),
+      kdfparams: new KdfParams(kdfparams)
+    })
+  }
+}
+
+export class Wallet extends Immutable.Record({
+  version: -1,
+  id: '',
+  address: '',
+  crypto: new Crypto()
+}) {
+  version: number
+  id: string
+  address: string
+  crypto: Crypto
+
+  constructor ({ crypto, ...rest }: Object = {}) {
+    super({ ...rest, crypto: new Crypto(crypto) })
```

Comment:
This technique is useful because it allows you to pass a a plain JS object to the constructor of an immutable record and that record will "automatically" convert pieces of that object to the appropriate records.

If we do not do this, which is also a valid approach, then we would have to pass in a JS object that already has certain keys set to the appropriate immutable records.